### DISCUSSION TITLE
Use refreshToken with cookies scheme

### DISCUSSION THREAD

**Discussion by mng1dev** [nuxt-community/auth-module#1547] [Q&A]
Hello,

I am currently working with a backend that provides users their access and refresh tokens via cookies. 
Since both `cookie` and `refresh` schemes extend the `local` scheme, there is not common `refreshToken` property.

Is there any way to use these schemes or should I just implement a custom one?

**Comment by skoenfaelt** [comment]
If i understand your question correctly [Refresh Scheme](https://auth.nuxtjs.org/schemes/refresh) should be the goal.

From the backend you need to send your token and refresh Token back to the frontend.

```js
auth: {
  strategies: {
    local: {
      scheme: 'refresh',
      token: {
        property: 'access_token',
        maxAge: 1800,
        // type: 'Bearer'  << default
      },
      refreshToken: {
        property: 'refresh_token',
        data: 'refresh_token',
        maxAge: 60 * 60 * 24 * 30
      },
      user: {... },
      endpoints: {...},
    }
  }
}
```

you may adjust the properties to your names, so frontend and backend align with each other. For more visit [the Docs](https://auth.nuxtjs.org/schemes/refresh#token)

**Comment by mng1dev** [comment]
The problem is that in my case tokens are not properties in a response payload, the response has a Set-Cookie header with authorization and refresh tokens.
